testjson/execution.go

```diff
@@ -344,6 +344,10 @@ func (p *Package) addTestEvent(event TestEvent) {
 	}
 
 	// the event.Action must be one of the three test end events
+	p.addEndTestEvent(event, tc)
+}
+
+func (p *Package) addEndTestEvent(event TestEvent, tc TestCase) {
```

Comment:
`gocyclo` forced this change on me. I'm personally not a fan of `gocyclo` and how it can force small fixes to be bundled with refactors, but hey, it's not my codebase.